include/sharg/parser.hpp

```diff
@@ -1025,6 +1023,92 @@ class parser
+     */
+    inline void verify_app_and_subcommand_names() const
+    {
+        // Before creating the detail::version_checker, we have to make sure that
+        // malicious code cannot be injected through the app name.
+        if (!std::regex_match(info.app_name, app_name_regex))
+        {
+            throw design_error{("The application name must only contain alpha-numeric characters or '_' and '-' "
+                                "(regex: \"^[a-zA-Z0-9_-]+$\").")};
+        }
+
+        for (auto & sub : this->subcommands)
+        {
+            if (!std::regex_match(sub, app_name_regex))
+            {
+                throw design_error{"The subcommand name must only contain alpha-numeric characters or '_' and '-' "
+                                   "(regex: \"^[a-zA-Z0-9_-]+$\")."};
+            }
+        }
+    }
+
+    /*!\brief Runs the version check if the user has not disabled it.
+     * \details
+     * If the user has not disabled the version check, the function will start a detached thread that will call the
+     * sharg::detail::version_checker and print a message if a new version is available.
+     */
+    inline void run_version_check()
+    {
+        detail::version_checker app_version{info.app_name, info.version, info.url};
+
+        if (app_version.decide_if_check_is_performed(version_check_dev_decision, version_check_user_decision))
+        {
+            // must be done before calling parse on the format because this might std::exit
+            std::promise<bool> app_version_prom;
+            version_check_future = app_version_prom.get_future();
+            app_version(std::move(app_version_prom));
+        }
+    }
+
+    /*!\brief Parses the command line arguments according to the format.
+     * \throws sharg::option_declared_multiple_times if an option that is not a list was declared multiple times.
+     * \throws sharg::user_input_error if an incorrect argument is given as (positional) option value.
+     * \throws sharg::required_option_missing if the user did not provide a required option.
+     * \throws sharg::too_many_arguments if the command line call contained more arguments than expected.
+     * \throws sharg::too_few_arguments if the command line call contained less arguments than expected.
+     * \throws sharg::validation_error if the argument was not excepted by the provided validator.
+     * \details
+     * This function calls the parse function of the format member variable.
+     */
+    inline void parse_format()
+    {
+        auto format_parse_fn = [this]<typename format_t>(format_t & f)
+        {
+            if constexpr (std::same_as<format_t, detail::format_tdl>)
+                f.parse(info, executable_name);
+            else
+                f.parse(info);
+        };
+
+        try
+        {
+            std::visit(std::move(format_parse_fn), format);
+        }
+        catch (sharg::too_many_arguments const & exception)
```

Comment:
A bit ugly. `too_many_arguments` is only thrown in `format_parse`, so that's fine.

It would be better if the format knew the subcommands, or had a pointer to the parser.
This might get addressed in the follow-up.